internal/statestore/public.go

```diff
@@ -36,14 +36,18 @@ type Service interface {
 	// DeleteTicket removes the Ticket with the specified id from state storage. This method succeeds if the Ticket does not exist.
 	DeleteTicket(ctx context.Context, id string) error
 
-	// IndexTicket indexes the Ticket id for the configured index fields.
+	// IndexTicket adds the ticket to the index.
 	IndexTicket(ctx context.Context, ticket *pb.Ticket) error
 
-	// DeindexTicket removes the indexing for the specified Ticket. Only the indexes are removed but the Ticket continues to exist.
+	// DeindexTicket removes specified ticket from the index. The Ticket continues to exist.
 	DeindexTicket(ctx context.Context, id string) error
 
-	// FilterTickets returns the Ticket ids for the Tickets meeting the specified filtering criteria.
-	FilterTickets(ctx context.Context, pool *pb.Pool, pageSize int, callback func([]*pb.Ticket) error) error
+	// GetIndexedIds returns the ids of all tickets currently indexed.
+	GetIndexedIds(ctx context.Context) (map[string]struct{}, error)
```

Comment:
I see we need a map on the receiving end for our use case - but simply looking at this API surface, the below seems more natural:
	GetIndexedIds(ctx context.Context) ([]string, error)
We can always convert it into a map at the receiving end.

Reply:
Both the interface implementation and the caller require operating on it as a map.  Converting it to a slice only to turn it back feels like a waste.  (both cycles, and more code)

Switched the name to GetIndexedIdSet to align more with what it's returning.  Is that acceptable?

Reply:
Designing an API for a single client use-case is a good approach. This approach is completely acceptable from say - helpers in same package where you return a type convenient for the caller as the helper is meant to be internal. 

For Interfaces that may be potentially implemented by other backends in future, focusing on more generic primitives helps. However, I do see renaming this slight improvement over the previous approach and my concern is not big enough to hold the PR.